roofit/roofitcore/src/RooDataHist.cxx

```diff
@@ -1142,6 +1142,49 @@ RooPlot *RooDataHist::plotOn(RooPlot *frame, PlotOpt o) const
   return RooAbsData::plotOn(frame,o) ;
 }
 
+
+void RooDataHist::weights(double* output, RooSpan<double const> xVals, bool correctForBinSize) 
+{
+  auto nEvents = xVals.size();
+
+  RooAbsBinning const& binning = *_lvbins[0];
+
+  if (binning.isUniform()) {
+    double minVal = binning.binLow(0);
+    double maxValFirstBin = binning.binHigh(0);
+
+    double oneOverBinWidth;
+    if (correctForBinSize) {
+      oneOverBinWidth = 1. / (maxValFirstBin - minVal);
+    } else {
+      oneOverBinWidth = 1.;
+    }
+
+    for(std::size_t i = 0; i < nEvents; ++i) {
+      auto binIndex = static_cast<int>((xVals[i] - minVal) * oneOverBinWidth);
+      output[i] = weight(binIndex) * oneOverBinWidth;
+    }
+  }
+  else {
+    std::vector<int> binIndices(nEvents);
+    binning.binNumbers(xVals.data(), binIndices.data(), nEvents);
```

Comment:
```suggestion
    assert(sizeof(int) < sizeof(double)) // make sure that the ouput buffer is large enough to contain the bin indices
    auto binIndices = reinterpret_cast<int*>(output + nEvents) - nEvents; // reuse the output buffer for bin indices
    std::fill(binIndices, binIndices + nEvents, 0); // output buffer for bin indices needs to be zero-initialized
    binning.binNumbers(xVals.data(), binIndices, nEvents);
```
We should avoid memory reallocations in PDF evaluation functions, because in a real fit they get called many times.

Reply:
`assert(sizeof(int) < sizeof(double))` will always be true, this assertion is useless. Also, use `static_assert` (compile time) to do this if you want this assertion.
Also, reinterpret_cast isn't very nice. Avoiding memory allocations is great, but remember that you should pay respect to the hardware alignment requirements. When you go from double to int, that's all fine, but it looks a bit dangerous.